keras_nlp/layers/transformer_decoder.py

```diff
@@ -266,6 +272,11 @@ def call(
         else:
             decoder_mask = tf.minimum(decoder_mask, causal_mask)
 
+        if self.normalize_first:
```

Comment:
We should not have variables that only exist inside concurrent if blocks, that's confusing. We should probably have something more like.

```
input_decoder_sequence = decoder_sequence
if self.normalize_first:
    decoder_sequence = self._decoder_attention_layernorm(decoder_sequence)

...

attention_output = input_decoder_sequence + self_attended
if not self.normalize_first:
    attention_output = self._decoder_attention_layernorm(attention_output)
```